src/Controller/Component/RequestHandlerComponent.php

```diff
@@ -177,7 +177,7 @@ public function startup(EventInterface $event): void
         }
 
         $isAjax = $request->is('ajax');
-        $controller->setRequest($request->withParam('isAjax', $isAjax));
+        $controller->setRequest($request->withAttribute('isAjax', $isAjax));
```

Comment:
These parameters could be in use by applications. We can likely migrate these with rector. I don't see an existing rector that can do this but I can try adding one.

Reply:
I hope it can be handled with rector. Would be real sad to be not be able to clean up cruft even in a major release.